service/src/main/java/dk/dbc/rr/oai/Config.java

```diff
@@ -0,0 +1,220 @@
+    private UriBuilder formatServiceUri;
+    private Client httpClient;
+    private Integer parallelFetch;
+    private Integer poolMinIdle;
+    private Integer poolMaxIdle;
+    private int fetchTimeoutInSeconds;
+
+    public Config() {
+        this(System.getenv());
+    }
+
+    public Config(Map<String, String> env) {
+        this.env = env;
+    }
+
+    public static Config of(String... envs) {
+        Map<String, String> env = Stream.of(envs)
+                .map(s -> s.split("=", 2))
+                .collect(toMap(a -> a[0], a -> a[1]));
+        return new Config(env);
+    }
+
+    @PostConstruct
+    public void init() {
+        log.info("Setting up config");
+
+        this.exposedUrl = getenv("EXPOSED_URL").get();
+        this.httpClient = getenv("USER_AGENT", "RawRepoOaiService/1.0")
+                .convert(userAgent -> ClientBuilder.newBuilder()
+                        .register((ClientRequestFilter) (ClientRequestContext context) ->
+                                context.getHeaders().putSingle("User-Agent", userAgent)
+                        )
+                        .register(new JacksonFeature())
+                        .build());
+        this.fetchTimeoutInSeconds = getenv("FETCH_TIMEOUT_IN_SECONDS").asInt()
+                .min(1)
+                .get();
+        this.parallelFetch = getenv("PARALLEL_FETCH").asInt()
+                .min(1)
+                .get();
+        this.poolMinIdle = getenv("POOL_MIN_IDLE").asInt()
+                .min(0)
+                .get();
+        this.poolMaxIdle = getenv("POOL_MAX_IDLE").asInt()
+                .min(1, "less that 1 whould create/destroy DOM Parser for every call")
+                .min(poolMinIdle + 1, "is should be more that POOL_MIN_IDLE")
+                .get();
+        this.formatServiceUri = getenv("RAWREPO_OAI_FORMATTER_SERVICE_URL")
+                .isNot("not empty", String::isEmpty)
+                .convert(UriBuilder::fromUri)
+                .path("api/format");
+    }
+
+    public String getExposedUrl() {
+        return exposedUrl;
+    }
+
+    public int getFetchTimeoutInSeconds() {
+        return fetchTimeoutInSeconds;
+    }
+
+    public UriBuilder getFormatServiceUri() {
+        return formatServiceUri.clone();
+    }
+
+    public Client getHttpClient() {
+        return httpClient;
+    }
+
+    public Integer getParallelFetch() {
+        return parallelFetch;
+    }
+
+    public Integer getPoolMaxIdle() {
+        return poolMaxIdle;
+    }
+
+    public Integer getPoolMinIdle() {
+        return poolMinIdle;
+    }
+
+    private static class FromEnv {
+
+        private final String name;
+        private final String value;
+
+        private FromEnv(String name, String value) {
+            this.name = name;
+            this.value = value;
+        }
+
+        private String get() {
+            return value;
+        }
+
+//        private FromEnv is(String test, Predicate<String> p) {
```

Comment:
get rid of this if it is not needed

Reply:
It will eventually, but not really rocket science to write again.